entrypoint.py

```diff
@@ -233,23 +239,62 @@
 git = git.Repo().git
 git.config('--global', 'user.name', os.environ['INPUT_GIT_USER_NAME'])
 git.config('--global', 'user.email', os.environ['INPUT_GIT_USER_EMAIL'])
-token  = os.environ['RELEASE_TOKEN']
-remote = 'https://' + token + '@github.com/' + os.environ['GITHUB_REPOSITORY']
+if deploy_key:
+    @contextmanager
+    def git_auth():
+        """
+        Temporarily set SSH credentials for Git. To be used as context manager.
+        """
+        (ssh_wrapper_fd, ssh_wrapper_path) = mkstemp(text=True)
+        try:
+            with NamedTemporaryFile() as identity_file:
+                with open(ssh_wrapper_fd, "w") as ssh_wrapper_file:
+                    ssh_wrapper_file.write('#!/bin/sh\n')
+                    ssh_wrapper_file.write(
+                        f'exec ssh -o StrictHostKeyChecking=no '
+                        f'-i {identity_file.name} $@')
+                    os.chmod(ssh_wrapper_path, 0o500)
+
+                identity_file.write(deploy_key.encode('utf-8'))
+                identity_file.flush()
+                os.environ['GIT_SSH'] = ssh_wrapper_path
+                try:
+                    yield
+                finally:
+                    del os.environ['GIT_SSH']
```

Comment:
It's probably better to use [`custom_environment()`](https://gitpython.readthedocs.io/en/stable/reference.html#git.cmd.Git.custom_environment) here.

Reply:
That's what I initially though as well, but the `mkdocs` invocation will also require Git credentials. That's why I went with this solution instead.